### PR TITLE
OTE_SDK. Increase code coverage. entities/model_template.py

### PR DESCRIPTION
**Pull request by saltykox**
Test list:
ote_sdk/ote_sdk/tests/entities/test_model_template.py

.yaml files copied from Pull Request [https://github.com/openvinotoolkit/training_extensions/pull/719/files](url)

Task: CVS-71811

### CODE REVIEW

**Comment by saltykox** [comment]
run ote_sdk tests

**Comment by saltykox** [comment]
run ote_sdk tests

**Comment by saltykox** [comment]
@LeonidBeynenson please review

**Comment by Ilya-Krylov** [comment]
please merge `develop` into your branch

File: ote_sdk/ote_sdk/tests/entities/test_model_template.py [hunk 1]
```diff
@@ -0,0 +1,1202 @@
...
+class TestTargetDevice:
+    @pytest.mark.priority_medium
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    def test_targetdevice(self):
+        """
+        <b>Description:</b>
+        Check TargetDevice IntEnum class elements
+        <b>Expected results:</b>
+        Test passes if TargetDevice IntEnum class length equal expected value and its elements have expected
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_model_template.py]:
```suggestion
        Test passes if TargetDevice IntEnum class length is equal to expected value and its elements have expected
```

File: ote_sdk/ote_sdk/tests/entities/test_model_template.py [hunk 2]
```diff
@@ -0,0 +1,1202 @@
...
+class TestModelOptimizationMethod:
+    @pytest.mark.priority_medium
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    def test_modeloptymizationmethod(self):
+        """
+        <b>Description:</b>
+        Check TestModelOptimizationMethod Enum class elements
+        <b>Expected results:</b>
+        Test passes if TestModelOptimizationMethod Enum class length, methods and attributes return expected values
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_model_template.py]:
```suggestion
        Test passes if ModelOptimizationMethod Enum class length, methods and attributes return expected values
```

File: ote_sdk/ote_sdk/tests/entities/test_model_template.py [hunk 3]
```diff
@@ -0,0 +1,1202 @@
...
+
+@pytest.mark.components(OteSdkComponent.OTE_SDK)
+class TestModelOptimizationMethod:
+    @pytest.mark.priority_medium
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    def test_modeloptymizationmethod(self):
+        """
+        <b>Description:</b>
+        Check TestModelOptimizationMethod Enum class elements
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_model_template.py]:
```suggestion
        Check ModelOptimizationMethod Enum class elements
```

File: ote_sdk/ote_sdk/tests/entities/test_model_template.py [hunk 4]
```diff
@@ -0,0 +1,1202 @@
...
+
+@pytest.mark.components(OteSdkComponent.OTE_SDK)
+class TestDatasetRequirements:
+    @pytest.mark.priority_medium
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    def test_datasetrequirements(self):
+        """
+        <b>Description:</b>
+        Check TestDatasetRequirements dataclass
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_model_template.py]:
```suggestion
        Check DatasetRequirements dataclass
```

File: ote_sdk/ote_sdk/tests/entities/test_model_template.py [hunk 5]
```diff
@@ -0,0 +1,1202 @@
...
+        <b>Expected results:</b>
+        Test passes if classes attribute of TestDatasetRequirements dataclass returns expected values
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_model_template.py]:
```suggestion
        Test passes if the classes attribute of DatasetRequirements dataclass returns expected values
```

**Comment by saltykox** [comment]
merged changes from "develop" branch

File: ote_sdk/ote_sdk/tests/entities/test_model_template.py [hunk 6]
```diff
@@ -0,0 +1,1202 @@
...
+    @pytest.mark.priority_medium
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    def test_tasktype_to_label_domain(self):
+        """
+        <b>Description:</b>
+        Check task_type_to_label_domain function
+        <b>Expected results:</b>
+        Test passes if task_type_to_label_domain function returns expected mapping of TaskType class element
+        to Domain class element
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_model_template.py:276]:
```suggestion
        to Domain class element expected values
```

File: ote_sdk/ote_sdk/tests/entities/test_model_template.py [hunk 7]
```diff
@@ -0,0 +1,1202 @@
...
+        expected values
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_model_template.py:277]:
```suggestion
```

File: ote_sdk/ote_sdk/tests/entities/test_model_template.py [hunk 8]
```diff
@@ -0,0 +1,1202 @@
...
+        """
+        <b>Description:</b>
+        Check data property of HyperParameterData class instance
+        <b>Expected results:</b>
+        Test passes if data property of HyperParameterData class instance has expected values
+        <b>Steps</b>
+        1. Check that data attribute of HyperParameterData object has empty dictionary before using load_parameters
+        method
+        2. Check scenario when path to config.yaml file specified in base_path parameter during HyperParameterData
+        object initiation
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_model_template.py:365]:
```suggestion
        object initialization
```

File: ote_sdk/ote_sdk/tests/entities/test_model_template.py [hunk 9]
```diff
@@ -0,0 +1,1202 @@
...
+    @pytest.mark.reqids(Requirements.REQ_1)
+    def test_hyperparameterdata_has_valid_configurable_parameters(self):
+        """
+        <b>Description:</b>
+        Check has_valid_configurable_parameters property of HyperParameterData class instance
+        <b>Expected results:</b>
+        Test passes if has_valid_configurable_parameters property of HyperParameterData class instance has
+        expected values
+        <b>Steps</b>
+        1. Check scenario when base_path parameter specified during HyperParameterData object initiation and
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_model_template.py:475]:
```suggestion
        1. Check scenario when base_path parameter specified during HyperParameterData object initialization and
```

File: ote_sdk/ote_sdk/tests/entities/test_model_template.py [hunk 10]
```diff
@@ -0,0 +1,1202 @@
...
+    @pytest.mark.reqids(Requirements.REQ_1)
+    def test_hyperparameterdata_has_overrides(self):
+        """
+        <b>Description:</b>
+        Check has_overrides property of HyperParameterData class instance
+        <b>Expected results:</b>
+        Test passes if has_overrides property of HyperParameterData class instance has expected values
+        <b>Steps</b>
+        1. Check scenario when HyperParameterData object has specified parameter_overrides dictionary
+        2. Check scenario when parameter_overrides dictionary not specified for HyperParameterData object
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_model_template.py:441]:
```suggestion
        2. Check scenario when parameter_overrides dictionary is not specified for HyperParameterData object
```

File: ote_sdk/ote_sdk/tests/entities/test_model_template.py [hunk 11]
```diff
@@ -0,0 +1,1202 @@
...
+        <b>Expected results:</b>
+        Test passes if has_valid_configurable_parameters property of HyperParameterData class instance has
+        expected values
+        <b>Steps</b>
+        1. Check scenario when base_path parameter specified during HyperParameterData object initiation and
+        model_template_path file specified in load_parameters method exists
+        2. Check scenario when __has_valid_configurable_parameters returns False when base_path parameter
+        not specified during HyperParameterData object initiation
+        3. Check scenario when __has_valid_configurable_parameters returns False when model_template_path file
+        specified in load_parameters method not exists
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_model_template.py:480]:
```suggestion
        specified in load_parameters method does not exist
```

File: ote_sdk/ote_sdk/tests/entities/test_model_template.py [hunk 12]
```diff
@@ -0,0 +1,1202 @@
...
+        <b>Description:</b>
+        Check has_valid_configurable_parameters property of HyperParameterData class instance
+        <b>Expected results:</b>
+        Test passes if has_valid_configurable_parameters property of HyperParameterData class instance has
+        expected values
+        <b>Steps</b>
+        1. Check scenario when base_path parameter specified during HyperParameterData object initiation and
+        model_template_path file specified in load_parameters method exists
+        2. Check scenario when __has_valid_configurable_parameters returns False when base_path parameter
+        not specified during HyperParameterData object initiation
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_model_template.py:478]:
```suggestion
        is not specified during HyperParameterData object initialization
```

**Comment by saltykox** [comment]
run ote_sdk tests

File: ote_sdk/ote_sdk/tests/entities/test_model_template.py [hunk 13]
```diff
@@ -0,0 +1,1202 @@
...
+        optional_parameters["exportable_code_paths"] = ExportableCodePaths.openvino
+        optional_parameters["task_type_sort_priority"] = 0
+        optional_parameters["gigaflops"] = 1
+        optional_parameters["size"] = 1024
+        return optional_parameters
+
+    @pytest.mark.priority_medium
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    def test_model_template_initiation(self):
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_model_template.py:836]:
```suggestion
    def test_model_template_initialization(self):
```

File: ote_sdk/ote_sdk/tests/entities/test_model_template.py [hunk 14]
```diff
@@ -0,0 +1,1202 @@
...
+        """
+        <b>Description:</b>
+        Check TestModelTemplate dataclass attributes
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_model_template.py:839]:
```suggestion
        Check ModelTemplate dataclass attributes
```

File: ote_sdk/ote_sdk/tests/entities/test_model_template.py [hunk 15]
```diff
@@ -0,0 +1,1202 @@
...
+        <b>Expected results:</b>
+        Test passes if classes attributes of ModelTemplate object return expected values
+        <b>Steps</b>
+        1. Check attributes of ModelTemplate object initiated with default values
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_model_template.py:843]:
```suggestion
        1. Check attributes of ModelTemplate object initialized with default values
```

File: ote_sdk/ote_sdk/tests/entities/test_model_template.py [hunk 16]
```diff
@@ -0,0 +1,1202 @@
...
+        2. Check attributes of ModelTemplate object initiated with fully specified values
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_model_template.py:844]:
```suggestion
        2. Check attributes of ModelTemplate object initialized with fully specified values
```

File: ote_sdk/ote_sdk/tests/entities/test_model_template.py [hunk 17]
```diff
@@ -0,0 +1,1202 @@
...
+        CommonMethods.check_model_attributes(
+            optional_parameters_model, self.optional_model_parameters()
+        )
+        assert default_model_template != optional_parameters_model
+        remove(optional_parameters_model.model_template_path)
+
+    @pytest.mark.priority_medium
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    def test_model_template_post_initiation(self):
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_model_template.py]:
```suggestion
    def test_model_template_post_initialization(self):
```

File: ote_sdk/ote_sdk/tests/entities/test_model_template.py [hunk 18]
```diff
@@ -0,0 +1,1202 @@
...
+        """
+        <b>Description:</b>
+        Check __post_init__ method of ModelTemplate dataclass
+        <b>Expected results:</b>
+        Test passes if __post_init__ method of ModelTemplate object raises ValueError exception for parameters violation
+        <b>Steps</b>
+        1. Check ValueError exception when initiated ModelTemplate has gRPC instantiation and "" gRPC address
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_model_template.py]:
```suggestion
        1. Check ValueError exception when initialized ModelTemplate has gRPC instantiation and "" gRPC address
```

File: ote_sdk/ote_sdk/tests/entities/test_model_template.py [hunk 19]
```diff
@@ -0,0 +1,1202 @@
...
+        2. Check ValueError exception when initiated ModelTemplate has CLASS instantiation and no entrypoints
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_model_template.py]:
```suggestion
        2. Check ValueError exception when initialized ModelTemplate has CLASS instantiation and no entrypoints
```

File: ote_sdk/ote_sdk/tests/entities/test_model_template.py [hunk 20]
```diff
@@ -0,0 +1,1202 @@
...
+        3. Check ValueError exception when initiated ModelTemplate has VISION task family and has no specified path to
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_model_template.py]:
```suggestion
        3. Check ValueError exception when initialized ModelTemplate has VISION task family and has no specified path to
```

File: ote_sdk/ote_sdk/tests/entities/test_model_template.py [hunk 21]
```diff
@@ -0,0 +1,1202 @@
...
+        config file
+        4. Check ValueError exception when initiated ModelTemplate with task family not equal to VISION and but has
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/entities/test_model_template.py]:
```suggestion
        4. Check ValueError exception when initialized ModelTemplate with task family is not equal to VISION and but has
```

File: ote_sdk/ote_sdk/tests/entities/test_model_template.py (hunk 1 above)
Comment (saltykox) [reply ote_sdk/ote_sdk/tests/entities/test_model_template.py]: Thanks, fixed test description

File: ote_sdk/ote_sdk/tests/entities/test_model_template.py (hunk 2 above)
Comment (saltykox) [reply ote_sdk/ote_sdk/tests/entities/test_model_template.py]: Thanks, fixed test description

File: ote_sdk/ote_sdk/tests/entities/test_model_template.py (hunk 3 above)
Comment (saltykox) [reply ote_sdk/ote_sdk/tests/entities/test_model_template.py]: Thanks, fixed test description

File: ote_sdk/ote_sdk/tests/entities/test_model_template.py (hunk 4 above)
Comment (saltykox) [reply ote_sdk/ote_sdk/tests/entities/test_model_template.py]: Thanks, fixed test description

File: ote_sdk/ote_sdk/tests/entities/test_model_template.py (hunk 5 above)
Comment (saltykox) [reply ote_sdk/ote_sdk/tests/entities/test_model_template.py]: Thanks, fixed test description

File: ote_sdk/ote_sdk/tests/entities/test_model_template.py (hunk 17 above)
Comment (saltykox) [reply ote_sdk/ote_sdk/tests/entities/test_model_template.py]: Thanks, renamed "test_model_template_post_initiation" to "test_model_template_post_initialization"

File: ote_sdk/ote_sdk/tests/entities/test_model_template.py (hunk 18 above)
Comment (saltykox) [reply ote_sdk/ote_sdk/tests/entities/test_model_template.py]: Thanks, fixed test description

File: ote_sdk/ote_sdk/tests/entities/test_model_template.py (hunk 19 above)
Comment (saltykox) [reply ote_sdk/ote_sdk/tests/entities/test_model_template.py]: Thanks, fixed test description

File: ote_sdk/ote_sdk/tests/entities/test_model_template.py (hunk 20 above)
Comment (saltykox) [reply ote_sdk/ote_sdk/tests/entities/test_model_template.py]: Thanks, fixed test description

File: ote_sdk/ote_sdk/tests/entities/test_model_template.py (hunk 21 above)
Comment (saltykox) [reply ote_sdk/ote_sdk/tests/entities/test_model_template.py]: Thanks, fixed test description

**Comment by Ilya-Krylov** [comment]
@saltykox I think you aware of possibility to simply click on 'Commit suggestion' and 'Add suggestion to batch'

**Comment by Ilya-Krylov** [comment]
run ote_sdk tests

**Pull request merged**